packages/nameguard-react/src/components/NameShield/index.tsx

```diff
@@ -1,50 +1,46 @@
 import React, { type ReactNode } from "react";
-import { ConsolidatedNameGuardReport, Rating } from "@namehash/nameguard";
+import { ConsolidatedNameGuardReport } from "@namehash/nameguard";
 import cc from "classcat";
 
 import { Tooltip } from "../Tooltip/Tooltip";
-import { Shield, ShieldSize } from "../Report/Shield";
-
-function textColor(rating?: Rating) {
-  switch (rating) {
-    case Rating.alert: {
-      return "text-red-600";
-    }
-    case Rating.pass: {
-      return "text-emerald-600";
-    }
-    case Rating.warn: {
-      return "text-amber-500";
-    }
-    default: {
-      return "text-gray-500";
-    }
-  }
-}
+import {
+  Shield,
+  ShieldSize,
+  getNameGuardRatingTextColors,
+} from "../Report/Shield";
+import { LoadingShield } from "../icons/LoadingShield";
+import { ErrorShield } from "../icons/ErrorShield";
 
 type NameShieldProps = {
   data?: ConsolidatedNameGuardReport;
   children?: ReactNode;
   disableHover?: boolean;
   size?: ShieldSize;
-} & React.ComponentProps<typeof Shield>;
+  error?: string;
```

Comment:
Why is error a string and what possible values might it have? Appreciate if we could refine and/or document this. It seems we're actually just using this as a boolean??

Reply:
It is really just a boolean used for `UnknownShieldIcon` rendering... Updated the prop type 👍🏼